src/hotspot/share/runtime/os.cpp

```diff
@@ -2173,17 +2173,19 @@ char* os::map_memory_to_file(size_t bytes, int file_desc, MEMFLAGS flag) {
 }
 
 char* os::attempt_map_memory_to_file_at(char* addr, size_t bytes, int file_desc, MEMFLAGS flag) {
-  char* result = pd_attempt_map_memory_to_file_at(addr, bytes, file_desc);
+  char* result = pd_attempt_map_memory_to_file_at(addr, bytes, file_desc, flag);
   if (result != nullptr) {
     MemTracker::record_virtual_memory_reserve_and_commit((address)result, bytes, CALLER_PC, flag);
   }
   return result;
 }
 
 char* os::map_memory(int fd, const char* file_name, size_t file_offset,
-                           char *addr, size_t bytes, bool read_only,
-                           bool allow_exec, MEMFLAGS flags) {
-  char* result = pd_map_memory(fd, file_name, file_offset, addr, bytes, read_only, allow_exec);
+                     char *addr, size_t bytes,
+                     MEMFLAGS flags,
+                     bool read_only,
+                     bool allow_exec) {
```

Comment:
The function was written with multiple parameters per line here, and then you changed it so that some of the params where placed on individual lines. This should likely be reverted.

Reply:
Fixed.